compiler/quilt/tools/store.py

```diff
@@ -314,10 +314,20 @@ def prune(self, objs=None):
 
 def parse_package_extended(name):
     hash = version = tag = None
+    package = name
     try:
         if ':' in name:
-            name, versioninfo = name.split(':', 1)
-            if ':' in versioninfo:
+            package, versioninfo = name.split(':', 1)
+            # in case name has team, versioninfo contains packagename itself
+            if '/' in versioninfo:
```

Comment:
I think it's best to use a regex at this point. Multiple levels of `if`s is getting pretty confusing.

Reply:
Agree, but there is an error raise in case it confused by parsing tag/version/hash. Thus, I should implement two regexs, one for parsing [team:]owner/package, second for the rest tag/version/hash or ignore it and make only one re and thus one raise exception?